### PR TITLE
Cleanup text-field

### PR DESCRIPTION
**Pull request by jpzwarte**
Since hint & error are now rendered by `<sl-form-field>`, the *only* thing the `<sl-text-field>` now contains is the input. This PR cleans up the component to remove the wrapper element (no longer necessary) and uses `delegatesFocus` for handling the focus.

### CODE REVIEW

File: packages/components/text-field/src/text-field.stories.ts [hunk 1]
```diff
@@ -189,62 +186,6 @@ export const Size: StoryObj = {
...
-      <sl-text-field id="input">
-        <div slot="hint-text">
-          Hint is an accessible way to provide <strong>additional information</strong> that might help the user
-        </div>
-      </sl-text-field>
-    </form>
-  `
-};
-
 export const ErrorMessageSizes: StoryObj = {
```
Comment (anna-lach) [inline packages/components/text-field/src/text-field.stories.ts:189]: Can you please update this story?
Error messages are not visible right now. :)

image

File: packages/components/text-field/src/text-field.stories.ts [hunk 2]
```diff
@@ -15,7 +15,6 @@ const sizes: TextFieldSize[] = ['md', 'lg'];
 export const API: StoryObj = {
```
Comment (anna-lach) [inline packages/components/text-field/src/text-field.stories.ts:15]: Can you please also improve min max length story? Validation message is not shown after validation clicked
image

Comment (anna-lach) [reply packages/components/text-field/src/text-field.stories.ts:15]: Also Pattern story:

image

Comment (anna-lach) [reply packages/components/text-field/src/text-field.stories.ts:15]: And also that story with Custom validation?
image

Comment (anna-lach) [inline packages/components/text-field/src/text-field.stories.ts:15]: If we want to keep those form stories, could you please update text fields there as well? :)

image

**Review by Diaan — APPROVED** [review]

File: packages/components/text-field/src/text-field.stories.ts (hunk 2 above)
Comment (jpzwarte) [reply packages/components/text-field/src/text-field.stories.ts]: I've updated the stories

Comment (jpzwarte) [reply packages/components/text-field/src/text-field.stories.ts]: I'll refactor the "Forms" stories later (as I mentioned in the checkbox PR).

**Pull request merged**